pkg/api/server/v1alpha2/results.go

```diff
@@ -67,6 +68,34 @@ func (s *Server) GetResult(ctx context.Context, req *pb.GetResultRequest) (*pb.R
 	return result.ToAPI(store), nil
 }
 
+// UpdateResult receives Result and FieldMask from client and uses them to update records in local Sqlite Server.
+func (s Server) UpdateResult(ctx context.Context, req *pb.UpdateResultRequest) (*pb.Result, error) {
+	// Retrieve result from database by name
+	tx := s.db.WithContext(ctx).Begin()
+	r, err := GetResultByName(tx, req.GetName())
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "failed to find a result: %v", err)
```

Comment:
What happens if we return here without a rollback? (I'm not sure what gorm's behavior is).

Reply:
Yeah it's a great point! I'll defer the rollback and commit operation.